rt-proj-02/src/cameras/orthographic.cpp

```diff
@@ -3,12 +3,45 @@
 
 namespace rt3 {
 
+OrthographicCamera::OrthographicCamera
+    (std::unique_ptr<Film> && f, Point3f lf, Point3f la, Vector3f up, ScreenWindow sw): 
+    Camera(std::move(f), lf, la, up, sw) {}
+
+
+Ray OrthographicCamera::generate_ray(int i, int j) {
+    auto [u2,v2] = get_uv(i,j);
+    Vector3f dir = w; // Expendable
+    return Ray{e + u * u + v * v, dir};
+}
+
 OrthographicCamera* create_orthographic_camera(
     const ParamSet& ps_camera,
     const ParamSet& ps_look_at, unique_ptr<Film>&& the_film
 ) {
+    ScreenWindow sw;
+
+    if(ps_camera.count("screen_window")) { // Try to retrieve screen window
+        sw = retrieve(ps_camera, "screen_window", ScreenWindow());
+    } else {
+        if(ps_camera.count("fovy")) { // If it doesn't exists, infer with the fovy
+            real_type fovy = retrieve(ps_camera, "fovy", real_type{});
+
+            // crazy math
+            real_type h = fabs(tan(fovy/2.0));
+            real_type asp = the_film->aspect_ratio();
+            sw = ScreenWindow(h * asp * -1, h * asp, -h, h);
+        } else { // If fovy doesn't exists, we have an error
+            RT3_ERROR("Could not infer Screen Window.");
+        }
+    }
 
-    return nullptr;
+    return new OrthographicCamera(
```

Comment:
In the Orthographic Camera we don't have a way to infer the Screen Window if it has not been provided. So it is enough to just return the new object of OrthographicCamera with the arguments retrieved with the ParamSet, as the following lines do. The lines above it are not necessary.